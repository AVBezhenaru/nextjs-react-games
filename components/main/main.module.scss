
.title {
  font-size: 66px;
  color: #D65B88;
}

.text {
  font-size: 22px;
  color: white;
}

.link {
  color: #D65B88;
}