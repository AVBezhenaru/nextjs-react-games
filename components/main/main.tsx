import Link from 'next/link';

import style from './main.module.scss';

const Main = () => (
  <>
    <h1 className={style.title}>Welcome</h1>
    <p className={style.text}>
      Go to{' '}
      <Link href="/login">
        <a className={style.link}>login</a>
      </Link>{' '}
      or{' '}
      <Link href="/registration">
        <a className={style.link}>registration</a>
      </Link>
    </p>
  </>
);

export default Main;
