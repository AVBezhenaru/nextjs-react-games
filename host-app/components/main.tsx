import style from "../styles/main.module.scss";
import Link from "next/link";

const Main = () => {
    return (
        <>
            <h1 className={style.title}>Welcome</h1>
            <p className={style.text}>Go to <Link href={"/login"}><a className={style.link}>login</a></Link> or <Link href={"/registration"}><a className={style.link}>registration</a></Link></p>
        </>
    );
};

export default Main;