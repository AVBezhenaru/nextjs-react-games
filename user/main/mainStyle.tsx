import styled from 'styled-components';

export const Section = styled.section`
  position: fixed;
  height: 100vh;
  margin: 0 auto;
  width: 100%;
  background-image: url('https://pro-theme.com/html/teamhost/assets/img/bg-first-screen.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  overflow: hidden;

  :before {
    content: '';
    display: block;
    width: 100%;
    height: 100%;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.69);
  }
`;

export const DivForm = styled.div`
  position: relative;
  margin: 10% auto;
  padding: 20px;
  max-width: 400px;
  height: 250px;
  background-color: rgba(255, 250, 246, 0.98);
  border-radius: 20px;
`;

export const H1 = styled.h1`
  font-size: 60px;
  color: #ff6927;
  text-align: center;
  margin: 30px 0;
`;

export const P = styled.p`
  display: flex;
  flex-direction: row;
  gap: 50px;
  font-size: 32px;
  color: red;
`;

export const InputBtn = styled.button`
  text-align: center;
  outline: none;
  border: none;
  padding: 20px 0;
  color: #ffffff;
  font-weight: 500;
  text-transform: uppercase;
  font-size: 20px;
  display: inline-block;
  width: 100%;
  position: relative;
  line-height: 16px;
  background-size: 300% 100%;
  border-radius: 50px;
  -o-transition: all 0.4s ease-in-out;
  transition: all 1s ease-in-out;
  background-image: linear-gradient(to right, #f5ce62, #e43603, #fa7199, #e85a19);
  box-shadow: 0 4px 15px 0 rgb(229 66 10 / 55%);

  :hover {
    color: white;
    transition: All 1s ease;
    background-position: 100% 0;
    outline: none;
  }
`;
