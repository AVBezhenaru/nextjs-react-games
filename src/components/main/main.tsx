/* eslint-disable react/no-children-prop */
import React from 'react';
import Link from 'next/link';

import { DivForm, H1, InputBtn, P, Section } from './mainStyle';

const Main = () => (
  <Section>
    <DivForm>
      <H1>Welcome</H1>
      <P>
        <Link href="/login">
          <InputBtn type="button" children="login" />
        </Link>
        <Link href="/registration">
          <InputBtn type="button" children="guest" />
        </Link>
      </P>
    </DivForm>
  </Section>
);

export { Main };
