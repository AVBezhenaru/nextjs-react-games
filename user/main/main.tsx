/* eslint-disable react/no-children-prop */
import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Link from 'next/link';

import Layout from '../../admin-navigate/layout/layout';

import { DivForm, H1, InputBtn, P, Section } from './mainStyle';

const Main = () => (
  <Section>
    <Router>
      <Routes>
        <Route path="/a" element={Layout()} />
      </Routes>
    </Router>
    <DivForm data-testid="form">
      <H1>Welcome</H1>
      <P>
        <Link href="/login">
          <InputBtn type="button" children="login" />
        </Link>
        <Link href="/registration">
          <InputBtn type="button" children="guest" />
        </Link>
      </P>
    </DivForm>
  </Section>
);

export { Main };
