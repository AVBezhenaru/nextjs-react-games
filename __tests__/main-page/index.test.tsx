/* eslint-disable no-undef */
import { render, screen, fireEvent } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import { Main } from '../../user/main/main';
import '@testing-library/jest-dom';

describe('Tests main page', () => {
  it('Test render', () => {
    render(<Main />);
    expect(screen.getByTestId('form')).toBeInTheDocument();
  });
  it('Snapshot main page', () => {
    const component = render(<Main />);
    expect(component).toMatchSnapshot();
  });
  it('Button registration test', () => {
    render(<Main />, { wrapper: MemoryRouterProvider });
    fireEvent.click(screen.getByText('guest'));
    expect(mockRouter.asPath).toEqual('/registration');
  });
  it('Button login test', () => {
    render(<Main />, { wrapper: MemoryRouterProvider });
    fireEvent.click(screen.getByText('login'));
    expect(mockRouter.asPath).toEqual('/login');
  });
});
