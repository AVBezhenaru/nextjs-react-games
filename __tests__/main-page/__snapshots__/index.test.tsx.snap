// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests main page Snapshot main page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="sc-bcXHqe kDbEkw"
      >
        <div
          class="sc-gswNZR fqyyjS"
          data-testid="form"
        >
          <h1
            class="sc-dkrFOg jYkYWt"
          >
            Welcome
          </h1>
          <p
            class="sc-hLBbgP cKNpgn"
          >
            <button
              class="sc-eDvSVe dHTmBL"
              type="button"
            >
              login
            </button>
            <button
              class="sc-eDvSVe dHTmBL"
              type="button"
            >
              guest
            </button>
          </p>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-bcXHqe kDbEkw"
    >
      <div
        class="sc-gswNZR fqyyjS"
        data-testid="form"
      >
        <h1
          class="sc-dkrFOg jYkYWt"
        >
          Welcome
        </h1>
        <p
          class="sc-hLBbgP cKNpgn"
        >
          <button
            class="sc-eDvSVe dHTmBL"
            type="button"
          >
            login
          </button>
          <button
            class="sc-eDvSVe dHTmBL"
            type="button"
          >
            guest
          </button>
        </p>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
